pyccl/base.py

```diff
@@ -0,0 +1,47 @@
+import sys
+from collections import OrderedDict
+import numpy as np
+
+
+def _to_hashable(obj):
+    """Make unhashable objects hashable in a consistent manner."""
+
+    if isinstance(obj, (int, float, str)):
+        # Strings and Numbers are hashed directly.
+        return obj
+
+    elif hasattr(obj, "__iter__"):
+        # Encapsulate all the iterables to quickly discard as needed.
+
+        if isinstance(obj, np.ndarray):
+            # Numpy arrays: Convert the data buffer to a byte string.
+            return obj.tobytes()
+
+        elif isinstance(obj, dict):
+            # Dictionaries: Build a tuple from key-value pairs,
+            # where all values are converted to hashables.
+            out = dict.fromkeys(obj)
+            for key, value in obj.items():
+                out[key] = _to_hashable(value)
+            # Sort unordered dictionaries for hash consistency.
+            if isinstance(obj, OrderedDict):
+                return tuple(obj.items())
```

Comment:
Wait, shouldn't you be using `out` here?

Reply:
Correct. Fixed.